src/http/nodegen/middleware/applicationMiddleware.ts

```diff
@@ -44,18 +43,42 @@ export const requestParser = (app: express.Application): void => {
 
   // parse the body
   app.use(express.urlencoded({ extended: false }));
-
-  app.use(requestIp.mw());
 };
 
+const ipHeaders = [
+  'x-client-ip',
+  'x-forwarded-for',
+  'cf-connecting-ip',
+  'do-connecting-ip',
+  'fastly-client-ip',
+  'true-client-ip',
+  'x-real-ip',
+  'x-cluster-client-ip',
+  'x-forwarded',
+  'forwarded-for',
+  'forwarded',
+  'x-appengine-user-ip',
+  'cf-pseudo-ipv4',
+];
+
+const getIpHeader = (req: express.Request): string => {
+  for (const header of ipHeaders) {
+    if (req[header]?.length) {
+      return req[header];
+    }
+  }
+
+  return req.socket.remoteAddress;
+}
```

Comment:
Does this work for ipv6 and also for localhost?

Reply:
Ipv6 yes, for localhost you just see ::1 (if that's what you mean).

The socket remote address is always present and it's the full IP of the incoming request - the only reason we also want to check any other headers is because that IP doesn't always come from the client but maybe a load balancer instead, so we check to see if the middleman saved the ip somewhere else.